src/main/java/com/esri/samples/na/closest_facility/ClosestFacilitySample.java

```diff
@@ -0,0 +1,211 @@
+      routeSymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, 0xFF0000FF, 2.0f);
+
+      // place incident were user clicks and display route to closest facility
+      mapView.setOnMouseClicked(e -> {
+        // check that the primary mouse button was clicked
+        if (e.getButton() == MouseButton.PRIMARY && e.isStillSincePress()) {
+          // show incident to the mapview
+          Point mapPoint = mapView.screenToLocation(new Point2D(e.getX(), e.getY()));
+          incidentPoint = new Point(mapPoint.getX(), mapPoint.getY(), spatialReference);
+          Graphic graphic = new Graphic(incidentPoint, incidentSymbol);
+          graphicsOverlay.getGraphics().add(graphic);
+
+          createParametersAndSolveRoute();
+        }
+      });
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Creates points around the San Diego region that will be used to create facilities and Graphics of those facilities.
+   */
+  private void createFacilitiesAndGraphics() {
+    // List of facilities to be placed around San Diego area
+    List<Point> facilityPoints = new ArrayList<>();
+    facilityPoints.add(new Point(-1.3042129900625112E7, 3860127.9479775648, spatialReference));
+    facilityPoints.add(new Point(-1.3042193400557665E7, 3862448.873041752, spatialReference));
+    facilityPoints.add(new Point(-1.3046882875518233E7, 3862704.9896770366, spatialReference));
+    facilityPoints.add(new Point(-1.3040539754780494E7, 3862924.5938606677, spatialReference));
+    facilityPoints.add(new Point(-1.3042571225655518E7, 3858981.773018156, spatialReference));
+    facilityPoints.add(new Point(-1.3039784633928463E7, 3856692.5980474586, spatialReference));
+    facilityPoints.add(new Point(-1.3049023883956768E7, 3861993.789732541, spatialReference));
+
+    // image for displaying facility
+    String facilityUrl = "http://static.arcgis.com/images/Symbols/SafetyHealth/Hospital.png";
+    PictureMarkerSymbol facilitySymbol = new PictureMarkerSymbol(facilityUrl);
+    facilitySymbol.setHeight(30);
+    facilitySymbol.setWidth(30);
+
+    List<Graphic> graphics = graphicsOverlay.getGraphics();
+    // list of facilities that will be needed to solve our closest facility route task
+    facilities = new ArrayList<>();
+    facilityPoints.forEach(facilityPoint -> {
+      facilities.add(new Facility(facilityPoint));
+      graphics.add(new Graphic(facilityPoint, facilitySymbol));
+    });
+  }
+
+  /**
+   * Adds facilities(hospitals) and user's incident(black cross) to closest facility parameters which will be used to 
+   * display the closest route from the user's incident to its' nearest facility.
+   */
+  private void createParametersAndSolveRoute() {
+    // parameters used to find closest facility to incident
+    final ListenableFuture<ClosestFacilityParameters> parameters = task.createDefaultParametersAsync();

```

Comment:
Do we have to create the default parameters every time or can we load this early and reuse it?